src/scripts/main.js

```diff
@@ -1,3 +1,244 @@
 'use strict';
 
-// write your code here
+const FIELD_SIZE = 4;
+let gameField;
+let score;
+
+const fieldCells = document.querySelectorAll('td');
+const gameScore = document.querySelector('.game-score');
+const startButton = document.querySelector('.start');
+const startMessage = document.querySelector('.message-start');
+const winMessage = document.querySelector('.message-win');
+const loseMessage = document.querySelector('.message-lose');
+
+window.addEventListener('load', () => {
+  initializeGame();
+});
+
+function initializeGame() {
+  startButton.addEventListener('click', () => {
+    startButton.className = 'button restart';
+    startButton.textContent = 'Restart';
+    startMessage.classList.add('hidden');
+    winMessage.classList.add('hidden');
+    loseMessage.classList.add('hidden');
+
+    gameField = Array(FIELD_SIZE).fill(0)
+      .map(row => Array(FIELD_SIZE).fill(0));
+    score = 0;
+
+    for (let i = 0; i < FIELD_SIZE; i++) {
+      for (let j = 0; j < FIELD_SIZE; j++) {
+        const currentCell = fieldCells[i * FIELD_SIZE + j];
+
+        currentCell.id = (`${i}-${j}`);
+
+        const value = gameField[i][j];
+
+        updateCell(currentCell, value);
+      }
+    }
+
+    document.addEventListener('keyup', manageGame);
+
+    addRandomCell();
+    addRandomCell();
+  });
+}
+
+function manageGame(e) {
+  if (e.code === 'ArrowLeft') {
+    slideHorizontal('left');
+    addRandomCell();
+  } else if (e.code === 'ArrowRight') {
+    slideHorizontal('right');
+    addRandomCell();
+  } else if (e.code === 'ArrowUp') {
+    slideVertical('up');
+    addRandomCell();
+  } else if (e.code === 'ArrowDown') {
+    slideVertical('down');
+    addRandomCell();
+  }
+
+  gameScore.textContent = score;
+  hasWon();
+  hasLost();
+}
+
+function hasWon() {
+  const cells = document.querySelectorAll('td');
+
+  for (const cell of cells) {
+    if (cell.classList.contains('field-cell--2048')) {
+      winMessage.classList.remove('hidden');
+
+      document.removeEventListener('keyup', manageGame);
+    }
+  }
+}
+
+function hasLost() {
+  if (
+    !canMoveHorizontally('left')
+    && !canMoveHorizontally('right')
+    && !canMoveVertically('up')
+    && !canMoveVertically('down')) {
+    loseMessage.classList.remove('hidden');
+
+    document.removeEventListener('keyup', manageGame);
+  }
+}
+
+function updateCell(cell, value) {
+  cell.textContent = '';
+  cell.classList.value = '';
+  cell.classList.add('field-cell');
+
+  if (value > 0) {
+    cell.textContent = value;
+    cell.classList.add(`field-cell--${value}`);
+  }
+}
+
+function hasEmptyCell() {
+  for (let i = 0; i < FIELD_SIZE; i++) {
```

Comment:
is it possible to simplify like `return gameField.some(row => row.includes(0));` ?